src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,121 @@
+package duke;
+
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * Represents the user interface of the Duke program, dealing with input and output.
+ */
+public class Ui {
+
+    private Scanner sc;
+    private String tab = "     ";
+    private String line = "____________________________________________________________";
+
+    /**
+     * Initializes a newly created Ui with a Scanner.
+     */
+    public Ui() {
+        this.sc = new Scanner(System.in);
+    }
+
+    /**
+     * Returns the user input.
+     *
+     * @return user input.
+     */
+    public String getInput() {
+        return sc.nextLine();
+    }
+
+    /**
+     * Greets the user.
+     */
+    public void greet() {
+        String tab = "     ";
+        chatPrint("Hello! I'm Duke\n" +
+                tab + "What can I do for you?");
+    }
+
+    /**
+     * Prints a farewell message to the user.
+     */
+    public void farewell() {
+        chatPrint("Bye. Hope to see you again soon!");
+    }
+
+    /**
+     * Prints the set output whenever the list command is called.
+     *
+     * @param tasks list of tasks.
+     */
+    public void onList(List<Task> tasks) {
```

Comment:
onList can be a bit misleading, can use printList/printCurrentList

Reply:
I see. That's a good idea.